tests/cpgrid/zoltan_test.cpp

```diff
@@ -115,9 +115,11 @@ BOOST_AUTO_TEST_CASE(zoltan)
 #endif
             grid.createCartesian(dims, size);
 
+        MPI_Barrier(MPI_COMM_WORLD);
```

Comment:
Doesn't hurt, but it is needed?

Reply:
It is needed. As mentioned above. At the same time, it was in the middle of us trying to figure out the Zoltan partition problem, it killed a little while for me to figure out. 

```MPI_Barrier is because when I run this test in some cluster with two processes, one process is still creating the grid, and the other one is already beginning to partition. So segmentation fault and the running gets stuck.```